File: src/main/java/testCodi/a03FrogJmp/frog_jmp.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(x, y, d)
  # write your code in Ruby 2.2
  if x == d
    0
  end
  candidate = (y - x).floor / d
  if x + candidate * d >= y
    candidate
  else
    candidate + 1
  end
end

File: src/main/java/testCodi/a03FrogJmp/frog_jmp.go
package a03frogjmp

// you can also use imports, for example:
// import "fmt"
// import "os"

// you can write to stdout for debugging purposes, e.g.
// fmt.Println("this is a debug message")

func SolutionFrogJmp(X int, Y int, D int) int {
	// write your code in Go 1.4
    if X == Y {
        return 0
    }
    var candidate = (Y - X) / D
    if X + candidate * D >= Y {
        return candidate
    } else {
        return candidate + 1
    }
}
